creator-node/src/fileManager.js

```diff
@@ -95,17 +96,55 @@ async function saveFileToIPFSFromFS (req, srcPath, fileType, sourceFile, transac
 }
 
 /**
- * Save file to disk given IPFS multihash, and ensure availability.
- * @notice This will only work for non-dir files
- * Steps:
- *  - If file already stored on disk, return immediately and store to disk.
- *  - If file not already stored, fetch from IPFS and store to disk.
- *    - If multihash available on local ipfs node, retrieve file.
- *    - If multihash not available locally, fetch file from IPFS.
- *  - If file is not available via IPFS try other cnode gateways for user's replica set.
- *  - Add file to local ipfs node if not already there.
+ * Given a CID, saves the file to disk. Steps to achieve that:
+ * 1. do the prep work to save the file to the local file system including
+ * creating directories, changing IPFS gateway urls before calling _saveFileForMultihash
+ * 2. attempt to fetch the CID from a variety of sources
+ * 3. throws error if failure, couldn't find the file or file contents don't match CID,
+ * returns expectedStoragePath if successful
+ * @param {Object} req request object
+ * @param {String} multihash IPFS cid
+ * @param {String} expectedStoragePath file system path similar to `/file_storage/Qm1`
+ *                  for non dir files and `/file_storage/Qmdir/Qm2` for dir files
+ * @param {Array} gatewaysToTry List of gateway endpoints to try
  */
-async function saveFileForMultihash (req, multihash, expectedStoragePath, gatewaysToTry = []) {
+async function saveFileForMultihash (req, multihash, expectedStoragePath, gatewaysToTry) {
+  const storagePath = req.app.get('storagePath') // should be like `/file_storage'
+  const filePath = expectedStoragePath.replace(storagePath, '') // should be `/Qm1/Qm2` for dir and `/Qm1` for non-dir
+
+  // will be modified to directory compatible route later if directory
+  let gatewayUrlsMapped = gatewaysToTry.map(endpoint => `${endpoint.replace(/\/$/, '')}/ipfs/`)
```

Comment:
instead of this `.replace(/\/$/, '')`, can we not use path.join or path.resolve like we do elsewhere?

Reply:
no we can't actually. path is meant for directory paths and the url module parses url and replaces routes. it doesn't join routes together. we'd need to install something like https://github.com/jfromaniello/url-join for that. but the more i think about it, probably makes sense to install

Reply:
gotcha - yeah prob doesn't need to happen in this PR but do you want to just add a TODO above this line pointed to that package so we don't forget?

Reply:
okay todo added